uktrade/data-hub-frontend
# Add financial year filter for investment projects

## Description of change

Adds a "Financial year" filter to the investment projects page (/investments/projects).

The financial year is determined on the api with this logic:
- Projects in the Prospect stage show under all financial years after they are created
- Non-prospect projects use the actual land date is it exists, falling back to estimated land date

I have also refactored the investment collection list page tests to follow the new pattern impleneted on the other collection lists.

See: https://uktrade.atlassian.net/browse/CRMD-42

## Test instructions

Go to /investments/projects - use the financial year filter - the url should be updated, the results should be filtered and a chip should appear.

## Screenshots

![financial-year-filter](https://user-images.githubusercontent.com/1234577/129865155-03170e8e-3950-4ea4-bd9c-8d6b24f4d054.png)

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
